Add Luminances and Scheme pages

attitude opened the pull request:
Add internal documentation & helper components to generate luminance scale & test schemes.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/contember/docs/52)
<!-- Reviewable:end -->

jonasNobile commented on src/theming/Luminance/Chart.tsx:
```diff
@@ -0,0 +1,169 @@
+ChartJS.register(CategoryScale, LineElement, PointElement, LinearScale, Title);
+
+const COLORS_COUNT = 41;
+const COLORS_RANGE = Array.apply(null, Array(COLORS_COUNT))
+  .map((_, index) => index)
+  .reverse();
```
```suggestion
const COLORS_RANGE = Array.from({length: COLORS_COUNT}, (_, index) => index).reverse()
```

jonasNobile commented on src/theming/Luminance/Chart.tsx:
```diff
@@ -0,0 +1,169 @@
+);
+
+const count = 41;
+const range = Array.apply(null, Array(count))
+  .map((_, index) => index)
+  .reverse();
```
```suggestion
const range = Array.from({length: count}, (_, index) => index).reverse()
```

jonasNobile commented on src/theming/Luminance/Chart.tsx:
```diff
@@ -0,0 +1,169 @@
+          top: 0
+        }}
+      >
+        {"\n"}
```
```suggestion
        <br/>
```

jonasNobile commented on src/theming/Luminance/Chart.tsx:
```diff
@@ -0,0 +1,169 @@
+            .map((number) => Math.round(number * 10000) / 10000)
+            .filter((number, index) => index % 5 === 0)
+        )}
+        {"\n"}
```
```suggestion
        <br />
```

jonasNobile commented on src/theming/Helpers/rangeOfNumbers.ts:
```diff
@@ -0,0 +1,6 @@
+export function rangeOfNumbers (start: number, end: number, step: number = 1) {
```
```suggestion
export function rangeOfNumbers(start: number, end: number, step: number = 1) {
```

VojtaStanek commented:
The `rangeOfNumbers` func is:
a) too complicated
b) wrong (eg. why round?)

attitude commented:
@VojtaStanek, `rangeOfNumbers()` does suit its needs. I refactored the code to be more readable and fixed use of `Math.round()` to `Math.floor()` method and confirmed it is working. Can you be now more specific about your concerns?

(Please note, there is only a slight change this method would be use otherwise as internal for the docs.)

Thanks

VojtaStanek commented:
Looks ok now.

jonasNobile reviewed (changes requested):
> Reviewed 28 of 28 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/contember/docs/52)* status: all files reviewed, 1 unresolved discussion (waiting on @attitude)

___
*[src/theming/Helpers/cssToSASS.ts, line 2 at r1](https://reviewable.io/reviews/contember/docs/52#-N0Fqsj3D6gidb9LYyqA:-N0Fqsj4-XVcyOBariwF:bz916y2) ([raw file](https://github.com/contember/docs/blob/b41d84d8cfbc9c0f28071672ddcc2f42f6961d3d/src/theming/Helpers/cssToSASS.ts#L2)):*
> ```XML
> export function cssToSASS(css: string) {
>   return css.replace(/;/g, "").replace(/\{/g, "").replace(/\}/g, "").trim()
> ```

OR

```ts
css.replaceAll(";", "").replaceAll("{", "").replaceAll("}", "").trim()
```



_Suggestion:_
```XML
  return css.replace(/[;{}]/g, "").trim()
```


<!-- Sent from Reviewable.io -->

attitude reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/contember/docs/52)* status: 5 of 28 files reviewed, 1 unresolved discussion (waiting on @jonasNobile)

___
*[src/theming/Helpers/cssToSASS.ts, line 2 at r1](https://reviewable.io/reviews/contember/docs/52#-N0Fqsj3D6gidb9LYyqA:-N0FxuYh8WkafLsW17s2:b-896fix) ([raw file](https://github.com/contember/docs/blob/b41d84d8cfbc9c0f28071672ddcc2f42f6961d3d/src/theming/Helpers/cssToSASS.ts#L2)):*
<details><summary><i>Previously, jonasNobile (Jonáš Nobile) wrote…</i></summary><blockquote>

OR

```ts
css.replaceAll(";", "").replaceAll("{", "").replaceAll("}", "").trim()
```
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

jonasNobile reviewed (approved):
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/contember/docs/52)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @attitude)



<!-- Sent from Reviewable.io -->
